yt_dlp/extractor/generic.py

```diff
@@ -2184,12 +2186,30 @@ def report_detected(self, name, num=1, note=None):
 
         self._downloader.write_debug(f'Identified {num} {name}{format_field(note, None, "; %s")}')
 
-    def _fragment_query(self, url):
+    def _extra_manifest_info(self, info, manifest_url, is_info_dict=True):
+        extra_info = {}
+
         if self._configuration_arg('fragment_query'):
-            query_string = urllib.parse.urlparse(url).query
+            query_string = urllib.parse.urlparse(manifest_url).query
             if query_string:
-                return {'extra_param_to_segment_url': query_string}
-        return {}
+                extra_info['extra_param_to_segment_url'] = query_string
+
+        hex_or_none = lambda x: x if re.fullmatch(r'(?i)(0x)?[\da-f]+', x) else None
+        extra_info['hls_aes'] = traverse_obj(self._configuration_arg('hls_key', []), {
+            'uri': (0, {url_or_none}),
+            'key': (0, {hex_or_none}),
+            'iv': (1, {hex_or_none}),
+        }) or None
+
+        info.update(extra_info)
```

Comment:
```suggestion
        is_hex = lambda _, v: re.fullmatch(r'(?i)(0x)?[\da-f]+', v)
        info['hls_aes'] = traverse_obj(self._configuration_arg('hls_key'), {
            'uri': (0, {url_or_none}), 'key': (0, is_hex), 'iv': (1, is_hex),
        })
```

Reply:
`is_hex` doesn't work because the filter function branches out after we've already returned the value, so it's filtering nothing

```py
>>> is_hex = lambda _, v: re.fullmatch(r'(?i)(0x)?[\da-f]+', v)
>>> traverse_obj(['1a', '2b'], {'uri': (0, {url_or_none}), 'key': (0, is_hex), 'iv': (1, is_hex)})
{}
```
it would "work" like this
```py
>>> traverse_obj(['1a', '2b'], {'uri': (0, {url_or_none}), 'key': is_hex, 'iv': is_hex})
{'key': ['1a', '2b'], 'iv': ['1a', '2b']}
>>> traverse_obj(['1a', '2b'], {'uri': (0, {url_or_none}), 'key': is_hex, 'iv': is_hex}, get_all=False)
{'key': '1a', 'iv': '1a'}

```
but that's not useful because then we can't specify which value is the key and which is the iv